### PR TITLE
fix `upsert` index of zero

### LINKED ISSUE #7347: `[] | upsert 1 'foo'` produces bad error
### Describe the bug

See title.

### How to reproduce

```
〉[] | upsert 1 'foo'
Error: nu::shell::access_beyond_end (link)

  × Row number too large (max: 18446744073709551615).
   ╭─[entry #10:1:1]
 1 │ [] | upsert 1 'foo'
   ·             ┬
   ·             ╰── index too large (max: 18446744073709551615)
   ╰────

```

### Expected behavior

The error should say that upserting beyond 1 + the list's length doesn't work, instead of giving an overflowed "index too large"

### Screenshots

_No response_

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.72.0                                   |
| branch             |                                          |
| commit_hash        | 8fa965118c2e5a0beaa528d07eaf9b33d96779cb |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.65.0 (897e37553 2022-11-02)      |
| rust_channel       | stable-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.65.0 (4bc8f24d3 2022-10-20)      |
| pkg_version        | 0.72.0                                   |
| build_time         | 2022-11-29 18:11:35 +00:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |                                          |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kangaxx-0**
# Description

Try to fix #7347, this pr does not fix the error message to `upserting beyond 1 + the list's length doesn't work,`, I feel this is still not explicit or general engouh, if we want to change, what would be the best message to be printed out ?

Add tests too

# User-Facing Changes

_(List of all changes that impact the user experience here. This helps us keep track of breaking changes.)_

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by rgwood** [comment]
I think an easier way to solve this might be to just remove the `- 1` 

for idx in 0..*val {
    if let Some(v) = input.next() {
        pre_elems.push(v);
    } else {
        return Err(ShellError::AccessBeyondEnd(idx, *span));
    }
}

**Comment by rgwood** [comment]
Thanks for the PR!

Could we do this in a simpler way by just removing `- 1` from this line?

https://github.com/nushell/nushell/blob/db6c804b17c59874c6a963dd300b85c6fab18260/crates/nu-command/src/filters/upsert.rs#L172

I think that should actually be:
```rust
return Err(ShellError::AccessBeyondEnd(idx, *span)); 
```

It seems like `N` in the `index too large (max: N)` error message is always off by 1. For example:
image

That should say `max: 1` not `max: 0`, because `1` works: 
image

**Comment by Kangaxx-0** [comment]
> Thanks for the PR!
> 
> Could we do this in a simpler way by just removing `- 1` from this line?
> 
> https://github.com/nushell/nushell/blob/db6c804b17c59874c6a963dd300b85c6fab18260/crates/nu-command/src/filters/upsert.rs#L172
> 
> I think that should actually be:
> 
> ```rust
> return Err(ShellError::AccessBeyondEnd(idx, *span)); 
> ```
> 
> It seems like `N` in the `index too large (max: N)` error message is always off by 1. For example: image
> 
> That should say `max: 1` not `max: 0`, because `1` works: image

Good catch! I've made the change

File: crates/nu-command/src/filters/upsert.rs [hunk 1]
```diff
@@ -168,8 +168,10 @@ fn upsert(
             for idx in 0..*val {
                 if let Some(v) = input.next() {
                     pre_elems.push(v);
+                } else if idx == 0 {
+                    return Err(ShellError::AccessEmptyContent(*span));
```
Comment (rgwood) [inline crates/nu-command/src/filters/upsert.rs:172]:
```suggestion
```
Do we still need this bit? I think it's better to show `max: 0` than `empty content`

**Review by rgwood — APPROVED** [review]

**Pull request merged**